nestkernel/source_table.cpp

```diff
@@ -273,6 +273,62 @@ nest::SourceTable::resize_sources( const thread tid )
   sources_[ tid ].resize( kernel().model_manager.get_num_synapse_prototypes() );
 }
 
+bool
+nest::SourceTable::source_should_be_processed_( const thread rank_start, const thread rank_end, const Source& source )
+{
+  if ( source.is_processed() or source.is_disabled() )
+  {
+    // looks like we've processed this already, let's continue
+    return false;
+  }
+
+  const thread source_rank = kernel().mpi_manager.get_process_id_of_node_id( source.get_node_id() );
+
+  // determine whether this thread is responsible for this part of
+  // the MPI buffer; if not we just continue with the next iteration
+  // of the loop
+  if ( source_rank < rank_start or source_rank >= rank_end )
+  {
+    return false;
+  }
+
+  return true; // if we reach this we should process it
+}
+
+bool
+nest::SourceTable::next_entry_has_same_source( const SourceTablePosition& current_position,
+  const Source& current_source ) const
+{
+  if ( ( current_position.lcid + 1
+           < static_cast< long >( sources_[ current_position.tid ][ current_position.syn_id ].size() )
+         and sources_[ current_position.tid ][ current_position.syn_id ][ current_position.lcid + 1 ].get_node_id()
+           == current_source.get_node_id() ) )
+  {
+    return true;
+  }
+
+  return false;
+}
+
+bool
+nest::SourceTable::previous_entry_has_same_source( const SourceTablePosition& current_position,
+  const Source& current_source ) const
```

Comment:
I would suggest to replace the entire code by

```c++
const auto prev_lcid = current_position.lcid - 1;

if ( prev_lcid < 0 )
{
  return false;
}
else
{
  const auto &prev_source = sources_[ current_position.tid ][ current_position.syn_id ][ prev_lcid ];
  return not prev_source.is_processed() and prev_source.get_node_id() == current_source.get_node_id();
}
```

Reply:
why not even more compact, like the next_entry function?